vmx/src/commit.rs

```diff
@@ -4,17 +4,40 @@
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
-// use std::fmt::{Display, Formatter};
+use bytecheck::CheckBytes;
+use rkyv::{
+    ser::serializers::{BufferScratch, CompositeSerializer, WriteSerializer},
+    ser::Serializer,
+    Archive, Deserialize, Serialize,
+};
 
 use uplink::ModuleId;
 
 use std::collections::BTreeMap;
+use std::fs::OpenOptions;
+use std::io::Read;
+use std::path::Path;
 
-use crate::error::Error::{self, SessionError};
+use crate::error::Error::{self, PersistenceError, SessionError};
 
 pub const COMMIT_ID_BYTES: usize = 32;
-
-#[derive(Clone, Copy, PartialOrd, Ord, PartialEq, Eq)]
+const SESSION_COMMITS_SCRATCH_SIZE: usize = 64;
+
+#[derive(
+    PartialEq,
+    Eq,
+    Archive,
+    Serialize,
+    CheckBytes,
+    Deserialize,
+    PartialOrd,
+    Ord,
+    Hash,
+    Clone,
+    Copy,
+)]
+#[archive(as = "Self")]
+#[repr(C)]
```

Comment:
Is it necessary to make these `#[repr(C]`?

Reply:
this was done because ModuleId has it as well, and these objects have identical structure, we could argue this is not needed here but it might be needed as commit ids could be exported in the future